packages/frontend/src/pages/Periods/Details/PeriodDetailsPage.tsx

```diff
@@ -75,6 +75,8 @@ const PeriodDetailPage = (): JSX.Element => {
   const [detailsLoaded, setDetailsLoaded] = React.useState<boolean>(false);
   const { path, url } = useRouteMatch();
 
+  PeriodDetailLoader();
+
```

Comment:
Lets just delete the PeriodDetailsLoader entirely, and just move the 2 relevant lines into the PeriodDetailPage component:
```
  const { location } = useHistory();
  useSinglePeriodQuery(periodId, location.key);
```

- then we can return `null` if the necessary data isn't loaded:
```
  if (!detailsLoaded || !period || !period.recievers) return null;
```

- then we can delete the useEffect entirely

Reply:
yeah, i wanted to do that but problem is that i am getting this typescript error:
![image](https://user-images.githubusercontent.com/11771927/167267762-a20527f4-b262-4a9d-8543-e19ef1dbfd4a.png)

Reply:
Just add null as a return type for the component:

```
const PeriodDetailPage = (): JSX.Element | null => {
```